aioesphomeapi/model.py

```diff
@@ -289,6 +317,15 @@ class ClimateState(EntityState):
     swing_mode = attr.ib(
         type=ClimateSwingMode, converter=ClimateSwingMode, default=ClimateSwingMode.OFF
     )
+    custom_fan_mode = attr.ib(
+        type=str, default=None
+    )
+    preset = attr.ib(
+        type=ClimatePreset, converter=ClimatePreset, default=ClimatePreset.HOME
```

Comment:
```suggestion
        type=ClimatePreset, converter=ClimatePreset, default=ClimatePreset.ECO
```

See comment at the top of the file:
```py
# All fields in here should have defaults set
# Home Assistant depends on these fields being constructible
# with args from a previous version of Home Assistant.
# The default value should *always* be the Protobuf default value
# for a field (False, 0, empty string, enum with value 0, ...)
```

Reply:
Shouldn't in that case HOME be the default value? HOME essentially means normal operation, a.k.a. no special preset. If we put ECO as default, what would happen with devices that don't even have ECO?

Reply:
That depends on the HA implementation.

The main problem is: if `preset` is not set in `api_connection.cpp`, or with older esphome client versions. `aioesphomeapi` will _receive_ that as `0` aka `ECO` mode (lack of a field in protobuf means use its 0 value). `default` is not really used here and only for HA to be able to restore entity infos.

ECO as 0 value _would_ be ok if the HA implementation checks if the preset does actually exist in `self.preset_modes` ([here](https://github.com/home-assistant/core/pull/48116/files#diff-d335f2f1f824951421ef024c3be5ef2aee032239bc4bbb723b8a5b54814ae889R281).

But that's not really a clean solution (and any other api clients would have to do the same). So yes it would be best if we encode `HOME` as 0 (would be an API protocol change, but I think it's ok if we change it soon)

Reply:
You can replicate it like this: use an old esphome version (1.18.0), or a climate device with neither `preset` nor `away` traits. Then HA will show the climate preset as `ECO`

Reply:
I too think if HOME should be default then move it to 0 on both sides. Since its not actually used yet we can safely change it on ESPHome side

Reply:
Ok, let's do that, I will have some time to make the changes next week.

Reply:
@jesserockz  @OttoWinter Changes to make HOME 0 are ready, please review